### PR TITLE
Add preflightvalidationocp to kmm pkg

### PR DESCRIPTION
**Pull request by cdvultur**
This PR adds the PreflightValidationOCP support pkg to eco-goinfra

### CODE REVIEW

File: pkg/kmm/preflightvalidationsocp.go [hunk 1]
```diff
@@ -0,0 +1,290 @@
+package kmm
```
Comment (trewest) [inline pkg/kmm/preflightvalidationsocp.go:1]: nit: I think the file name should be preflightvalidationocp.go

Comment (cdvultur) [reply pkg/kmm/preflightvalidationsocp.go]: There were some places where this was at plural. Still nice catch !

Thanks , Updated

File: pkg/kmm/preflightvalidationocp.go [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	var err error
+	if !builder.Exists() {
+		err = builder.apiClient.Create(context.TODO(), builder.Definition)
+	}
+
+	return builder, err
+}
+
```
Comment (trewest) [inline pkg/kmm/preflightvalidationocp.go:189]: Missing comment

File: pkg/kmm/preflightvalidationocp.go [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+	}
+
+	glog.V(100).Infof("Updating preflightvalidationocp %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	err := builder.apiClient.Update(context.TODO(), builder.Definition)
+
+	return builder, err
+}
+
```
Comment (trewest) [inline pkg/kmm/preflightvalidationocp.go:202]: Missing comment

File: pkg/kmm/preflightvalidationocp.go [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+	glog.V(100).Infof("Checking if preflightvalidationocp %s exists in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	var err error
+	builder.Object, err = builder.Get()
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
```
Comment (trewest) [inline pkg/kmm/preflightvalidationocp.go:216]: Missing comment

File: pkg/kmm/preflightvalidationocp.go [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+	if err != nil {
+		return builder, err
+	}
+
+	builder.Object = nil
+
+	return builder, err
+}
+
```
Comment (trewest) [inline pkg/kmm/preflightvalidationocp.go:239]: Missing comment

File: pkg/kmm/preflightvalidationocp.go [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+	// ApiClient to interact with the cluster.
+	apiClient *clients.Settings
+	// errorMsg is processed before the object is created or updated.
+	errorMsg string
+}
+
+// PreflightValidationOCPAdditionalOptions additional options for preflightvalidationocp object.
+type PreflightValidationOCPAdditionalOptions func(
+	builder *PreflightValidationOCPBuilder) (*PreflightValidationOCPBuilder, error)
+
```
[the remark this answers is not in the document]
Comment (cdvultur) [reply pkg/kmm/preflightvalidationocp.go:32]: Fixed, Thanks!

File: pkg/kmm/preflightvalidationocp.go (hunk 4 above)
Comment (cdvultur) [reply pkg/kmm/preflightvalidationocp.go:218]: Fixed, thanks!

File: pkg/kmm/preflightvalidationocp.go (hunk 3 above)
Comment (cdvultur) [reply pkg/kmm/preflightvalidationocp.go:203]: Fixed, thanks!

File: pkg/kmm/preflightvalidationocp.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (cdvultur) [reply pkg/kmm/preflightvalidationocp.go:32]: Sorry, it got masked by the function above, in my editor

**Review by trewest — APPROVED** [review]
/lgtm

**Review by achuzhoy — APPROVED** [review]

**Pull request merged**